listener.go

```diff
@@ -99,6 +92,15 @@ func (l *listener) setupConn(sess quic.Session) (tpt.Conn, error) {
 
 // Close closes the listener.
 func (l *listener) Close() error {
+	network := l.conn.LocalAddr().Network()
+	l.transport.connManager.mu.Lock()
+	switch network {
+	case "udp4":
```

Comment:
So, go has a habit of throwing away ip6/ip4 information and returning just "udp". We probably need to *remember* whether we're using IPv4/IPv6 in the listener itself.

Reply:
Since key of map is the connection itself, call `delete` for both `ipv4Conns` and `ipv6Conns` might be easier.